plugins/modules/ndo_tenant_span_session.py

```diff
@@ -0,0 +1,624 @@
+    description = module.params.get("description")
+    sources = module.params.get("sources")
+    admin_state = module.params.get("admin_state")
+    mtu = module.params.get("mtu")
+    destination_epg = module.params.get("destination_epg")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "monitoring_tenant", template_name, template_id)
+    mso_template.validate_template("monitoring")
+    object_description = "SPAN Session"
+    path = "/monitoringTemplate/template/spanSessions"
+    span_session_path = None
+    reference_collection = {
+        "epg": {
+            "name": "epgName",
+            "reference": "epgRef",
+            "type": "epg",
+            "template": "epgTemplateName",
+            "templateId": "epgTemplateId",
+            "schema": "epgSchemaName",
+            "schemaId": "epgSchemaId",
+        }
+    }
+
+    existing_span_sessions = mso_template.template.get("monitoringTemplate", {}).get("template", {}).get("spanSessions") or []
+
+    if name or uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_span_sessions,
+            [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+        )
+        if match:
+            span_session_path = "{0}/{1}".format(path, match.index)
+            set_template_and_references(mso_template, match.details, reference_collection)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = [set_template_and_references(mso_template, obj, reference_collection) for obj in existing_span_sessions]
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        destination_epg_uuid = get_epg_uuid(mso, mso_schema, destination_epg.get("epg"), destination_epg.get("epg_uuid"))
+
+        mso_values = dict(
+            name=name,
+            description=description,
+            destination=dict(
+                remote=dict(
+                    epgRef=destination_epg_uuid,
+                    spanVersion=destination_epg.get("span_version"),
+                    enforceSpanVersion=destination_epg.get("enforce_span_version"),
+                    destIPAddress=destination_epg.get("destination_ip"),
+                    srcIPPrefix=destination_epg.get("source_ip_prefix"),
+                    flowID=destination_epg.get("flow_id"),
+                    ttl=destination_epg.get("ttl"),
+                    dscp=TARGET_DSCP_MAP.get(destination_epg.get("dscp")),
+                ),
+                mtu=mtu,
+            ),
+        )
+        if admin_state is not None or sources is not None:
+            source_group = dict()
+            if admin_state is not None:
+                source_group["enableAdminState"] = False if admin_state == "disabled" else True
```

Comment:
Can you use `ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP` for the state to bool conversion?